pkg/controller/statefulset/stateful_pod_control.go

```diff
@@ -77,12 +85,36 @@ func (spc *realStatefulPodControl) CreateStatefulPod(set *apps.StatefulSet, pod
 		return err
 	}
 	// If we created the PVCs attempt to create the Pod
-	_, err := spc.client.CoreV1().Pods(set.Namespace).Create(pod)
+	newPod, err := spc.client.CoreV1().Pods(set.Namespace).Create(pod)
 	// sink already exists errors
 	if apierrors.IsAlreadyExists(err) {
 		return err
 	}
 	spc.recordPodEvent("create", set, pod, err)
+
+	// update PVC's ownerReference
+	if utilfeature.DefaultFeatureGate.Enabled(features.AutoCleanupPersistentVolumeClaim) {
+		val, ok := set.Annotations[AutoCleanupPersistentVolumeClaim]
+		if ok && val == "" {
+			var controllerKind = v1.SchemeGroupVersion.WithKind("Pod")
+			ordinal := getOrdinal(pod)
+			templates := set.Spec.VolumeClaimTemplates
+			claimName := fmt.Sprintf("%s-%s-%d", templates[0].Name, set.Name, ordinal)
+			claim, _ := spc.pvcLister.PersistentVolumeClaims(set.Namespace).Get(claimName)
+
+			patch := fmt.Sprintf(
+				`{"metadata":{"ownerReferences":[{"apiVersion":"%s","kind":"%s","name":"%s","uid":"%s","controller":true,"blockOwnerDeletion":true}],"uid":"%s"}}`,
+				controllerKind.GroupVersion().String(), controllerKind.Kind,
+				pod.GetName(), newPod.GetUID(), claim.GetUID())
+			_, err = spc.client.CoreV1().PersistentVolumeClaims(set.Namespace).Patch(
```

Comment:
Can the owner reference be set when the statefulset controller first creates the pvc? instead of patched later?

Reply:
Because the PVC is created before pod creation，I can not get the pod's UID when insert the ownerReference into PVC.

Reply:
@msau42 @zhangxiaoyu-zidif Shouldn't the ownerRef bear the UID of the Statefulset rather than that of the Pod?